### PR TITLE
Upgrade tests to be compatible with PHPUnit 5.7 and drop support for dead PHP versions

### PR DESCRIPTION
**Pull request by Milad**
I have done all the necessary changes to the tests so that they become compatible with PHPUnit 5.7 which also requires PHP 5.6+.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -7,3 +7,4 @@ package.xml
 composer.lock
 experiments/
 vendor/
+.idea
```
Comment (GrahamCampbell) [inline .gitignore]: 👎 This belongs in your global ignore file.

Comment (Milad) [inline .gitignore]: work in progress. I have created the pull request to run the pipelines at travis!

File: composer.json [hunk 2]
```diff
@@ -19,7 +19,7 @@
         "php": ">=5.3.9"
```
Comment (GrahamCampbell) [inline composer.json]: This needs upgrading too.

**Comment by GrahamCampbell** [comment]
👎 On these changes. They have not been made, because it's simply not needed to upgrade phpunit. It's more important to have wider PHP version support.

**Comment by Milad** [comment]
I think it's not a good idea to allow dead php versions hold us back. In my project I use PHPUnit 6 and I had to downgrade to PHPUnit 5.7 just to satisfy the widest possible PHP versions here.

**Comment by nrk** [comment]
@Milad I'm sorry to say this but you should have discussed these changes on a separate issue **prior** to actually investing your time in this PR. Unfortunately I can't accept it for the sole reason that in Predis 1.x we won't ever bump the minimum supported version of PHP and there's no reason to change this just for upgrading phpunit. It'll be a whole different matter for Predis 2.x as you can see in #335 but I still haven't decided which minimum target version of PHP will be supported in the end (most likely 7.x at this point, not really sure yet). That being said, I really don't see how a `require-dev` dependency in Predis can affect your whole project since phpredis is being vendored anyway by composer.

**Pull request closed without merging**